CRM/Contribute/BAO/Contribution/Utils.php

```diff
@@ -120,11 +120,6 @@ public static function processConfirm(
       $paymentParams['contributionTypeID'] = $contributionTypeId;
       $paymentParams['item_name'] = $form->_params['description'];
 
-      if ($contribution && $form->_values['is_recur'] && $contribution->contribution_recur_id
```

Comment:
I made a comment elsewhere about risk here - but I think maybe you have actually tested multiple flows from your enotice comment - if that is the case may it can go - but it would be good to confirm what flows you tested.

It would be better to get rid of it as it is a case of 'do something with unclear consequences' if it truly is safe